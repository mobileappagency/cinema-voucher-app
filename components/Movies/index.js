// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'
import {
  ScrollView,
  StyleSheet,
  View
} from 'react-native'
import type { ReduxState, TvShowsResults, TvShowsResult } from 'types'
import { fetchAllTvShows } from '../../actions'
import MoviePoster from '../MoviePoster'
import MoviePopup from '../MoviePopup'

type State = {
  popupIsOpen: boolean,
  movie: ?TvShowsResult
}

type Props = {
  fetchAllTvShows: Function,
  tvShows: TvShowsResults
}

class Movies extends Component<Props, State> {
  state = {
    popupIsOpen: false,
    movie: null
  }

  componentWillMount () {
    this.props.fetchAllTvShows()
  }

  openMovie = (movie: TvShowsResult) => {
    this.setState({
      popupIsOpen: true,
      movie
    })
  }

  closeMovie = () => {
    this.setState({
      popupIsOpen: false
    })
  }

  render () {
    const { tvShows } = this.props

    console.log(this.state.popupIsOpen, this.state.movie)

    return (
      <View style={styles.container}>
        <ScrollView contentContainerStyle={styles.scrollContent}>
          {tvShows.results.map((movie, index) =>
            <MoviePoster movie={movie} key={index} onOpen={() => this.openMovie(movie)} />)}
        </ScrollView>
        <MoviePopup
          movie={this.state.movie}
          isOpen={this.state.popupIsOpen}
          onClose={this.closeMovie}
          />
      </View>
    )
  }
}

const mapStateToProps = ({ tvShows }): ReduxState => {
  return { tvShows }
}

const styles = StyleSheet.create({
  container: {
    paddingTop: 20
  },
  scrollContent: {
    flexDirection: 'row',
    flexWrap: 'wrap'
  }
})

export default connect(mapStateToProps, { fetchAllTvShows })(Movies)
