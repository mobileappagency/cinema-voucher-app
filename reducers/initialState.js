// @flow
import type { TvShowsStore } from './tvShowsReducer'

type ReduxState = {
  tvShows: TvShowsStore
}

const initialState: ReduxState = {
  tvShows: { genres: [], results: [] }
}

export default initialState
export type { ReduxState }
