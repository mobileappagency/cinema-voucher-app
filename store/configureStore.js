import { createStore, applyMiddleware, compose } from 'redux'
import createSagaMiddleware, { END } from 'redux-saga'
import rootReducer from '../reducers'

import type { ActionCreator, Store } from 'redux'
import type { SagaMiddlewareOptions, Task } from 'redux-saga'
import type { ReduxState } from '../reducers/initialState'

type ReduxStore = Store<ReduxState, ActionCreator<*>> & { runSaga: Task<*> }

export default function configureStore (initialState: ReduxState): ReduxStore {
  const sagaMiddleware: SagaMiddlewareOptions<any> = createSagaMiddleware()
  const store = createStore(
    rootReducer,
    initialState,
    compose(
      applyMiddleware(
        sagaMiddleware
      )
    )
  )

  store.runSaga = sagaMiddleware.run
  store.close = () => store.dispatch(END)
  return store
}
