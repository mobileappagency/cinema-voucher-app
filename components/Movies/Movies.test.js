// @flow
import React from 'react'
import configureStore from 'redux-mock-store'

import renderer from 'react-test-renderer'

import initialState from '../../reducers/initialState'
import Movies from './'

describe('Movies', () => {
  let mockStore
  let store

  beforeEach(() => {
    initialState.tvShows.results.push({
      name: 'Blade Runner',
      poster_path: 'blade-runner.jpg',
      vote_average: 4.5,
      original_language: 'en-US',
      overview: '... text ...',
      vote_count: 10000
    })
    mockStore = configureStore([])
    store = mockStore(initialState)
  })

  it('matches current snapshot', () => {
    const rendered: ReactTestRenderer = renderer.create(<Movies store={store} />)

    expect(rendered).toMatchSnapshot()
  })
})
