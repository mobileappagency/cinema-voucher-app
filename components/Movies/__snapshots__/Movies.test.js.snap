// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies matches current snapshot 1`] = `
<View
  style={
    Object {
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 365,
            "marginBottom": 5,
            "marginLeft": 7.5,
            "opacity": 1,
            "width": 240,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w500/blade-runner.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 5,
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "fontFamily": "Avenir",
              "fontSize": 14,
              "marginTop": 4,
            }
          }
        >
          Blade Runner
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#BBBBBB",
              "fontFamily": "Avenir",
              "fontSize": 12,
              "lineHeight": 14,
            }
          }
        >
          4.5
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
